src/cascadia/Remoting/WindowManager.cpp

```diff
@@ -118,78 +118,60 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
-                // bubble that up because that's a real issue.
+                // We did not successfully ask the king what to do. This could
+                // be for many reasons. Most commonly, the monarch died as we
+                // were talking to it. That could be a RPC_SERVER_UNAVAILABLE_HR
+                // or RPC_CALL_FAILED_HR (GH#12666). We also saw a
+                // RPC_S_CALL_FAILED_DNE in GH#11790. Ultimately, if this is
+                // gonna fail, we want to just try again, regardless of the
+                // cause. That's why we're no longer checking what the exception
+                // was, we're just always gonna try again regardless.
                 //
-                // I'm checking both these here. I had previously got a
-                // RPC_S_CALL_FAILED about here once.
-                if (e.code() == RPC_SERVER_UNAVAILABLE_HR || e.code() == RPC_CALL_FAILED_HR)
-                {
-                    TraceLoggingWrite(g_hRemotingProvider,
-                                      "WindowManager_proposeToMonarch_kingDied",
-                                      TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE),
-                                      TraceLoggingKeyword(TIL_KEYWORD_TRACE));
-
-                    // We failed to ask the monarch. It must have died. Try and
-                    // find the real monarch. Don't perform an election, that
-                    // assumes we have a peasant, which we don't yet.
-                    _createMonarchAndCallbacks();
-                    // _createMonarchAndCallbacks will initialize _isKing
-                    if (_isKing)
-                    {
-                        // We became the king. We don't need to ProposeCommandline to ourself, we're just
-                        // going to do it.
-                        //
-                        // Return early, because there's nothing else for us to do here.
-                        TraceLoggingWrite(g_hRemotingProvider,
-                                          "WindowManager_proposeToMonarch_becameKing",
-                                          TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE),
-                                          TraceLoggingKeyword(TIL_KEYWORD_TRACE));
+                //  They hopefully just died here. That's okay, let's just go
+                // ask the next in the line of succession. At the very worst,
+                // we'll find _us_, (likely last in the line).
 
-                        // In WindowManager::ProposeCommandline, had we been the
-                        // king originally, we would have started by setting
-                        // this to true. We became the monarch here, so set it
-                        // here as well.
-                        _shouldCreateWindow = true;
-                        return;
-                    }
+                // If the monarch (maybe us) failed for _any other reason_ than

```

Comment:
Also, this comment doesn't seem to follow from the previous paragraph. We're eating all the exceptions, not bubbling any exceptions out!

Reply:
whoops, missed one.